### ISSUE TITLE
Crash when testing on device after migrating from Frameworks to XCFrameworks

### ISSUE THREAD

**Issue by Bawenang** [Carthage/Carthage#3186]
* carthage install method: [ ] .pkg, [x] homebrew, [ ] source
* `which carthage`: /usr/local/bin/carthage
* `carthage version`:  0.38.0
* `xcodebuild -version`: Xcode 12.4 Build version 12D4e
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`? Yes
* Are you using `--use-submodules`? No
* Are you using `--cache-builds`? Yes
* Are you using `--new-resolver`? No

**Cartfile**
```
github "ReactiveX/RxSwift" ~> 6.2
github "Swinject/Swinject" "2.6.2"
github "Swinject/SwinjectStoryboard" "2.2.0"
github "kishikawakatsumi/KeychainAccess"
github "malcommac/SwiftDate" ~> 6.0
github "malcommac/SwiftRichString"
github "SnapKit/SnapKit" ~> 5.0.0
github "hackiftekhar/IQKeyboardManager"
github "SwiftKickMobile/SwiftMessages" "8.0.3"
```

**Carthage Output**
```
carthage update --platform ios --no-use-binaries --cache-builds --use-xcframeworks
*** Fetching SwiftDate
*** Fetching SwiftRichString
*** Fetching RxSwift
*** Fetching KeychainAccess
*** Fetching SnapKit
*** Fetching IQKeyboardManager
*** Fetching Swinject
*** Checking out KeychainAccess at "v4.2.2"
*** Checking out SwinjectStoryboard at "2.2.0"
*** Checking out IQKeyboardManager at "v6.5.6"
*** Checking out Swinject at "2.6.2"
*** Checking out SwiftDate at "6.3.1"
*** Checking out RxSwift at "6.2.0"
*** Checking out SnapKit at "5.0.1"
*** Checking out SwiftRichString at "3.7.2"
*** Checking out SwiftMessages at "8.0.3"
*** No cache found for IQKeyboardManager, building with all downstream dependencies
*** No cache found for KeychainAccess, building with all downstream dependencies
*** No cache found for RxSwift, building with all downstream dependencies
*** No cache found for SnapKit, building with all downstream dependencies
*** No cache found for SwiftDate, building with all downstream dependencies
*** No cache found for SwiftMessages, building with all downstream dependencies
*** No cache found for SwiftRichString, building with all downstream dependencies
*** No cache found for Swinject, building with all downstream dependencies
*** xcodebuild output can be found in /var/folders/0t/xl5r053d2237cckk_ttg7slh0000gn/T/carthage-xcodebuild.DZ3b2u.log
*** Building scheme "IQKeyboardManager" in IQKeyboardManager.xcodeproj
*** Building scheme "IQKeyboardManagerSwift" in IQKeyboardManager.xcodeproj
*** Building scheme "KeychainAccess" in KeychainAccess.xcworkspace
*** Building scheme "RxBlocking" in Rx.xcworkspace
*** Building scheme "RxSwift" in Rx.xcworkspace
*** Building scheme "RxCocoa" in Rx.xcworkspace
*** Building scheme "RxRelay" in Rx.xcworkspace
*** Building scheme "RxTest" in Rx.xcworkspace
*** Building scheme "SnapKit" in SnapKit.xcworkspace
*** Building scheme "SwiftDate-iOS" in SwiftDate.xcodeproj
*** Building scheme "SwiftMessages" in SwiftMessages.xcodeproj
*** Building scheme "SwiftRichString-iOS" in SwiftRichString.xcworkspace
*** Building scheme "Swinject-iOS" in Swinject.xcodeproj
*** Building scheme "SwinjectStoryboard-iOS" in SwinjectStoryboard.xcodeproj
```

**Actual outcome**
Carthage crashed when testing the app on an actual device (it worked fine on simulators). The error was `Thread 1: EXC_BAD_ACCESS (code=2, address=0x256760eb0)`. It worked fine using FAT frameworks. The last texts on the consoles were these (I don't know whether these are relevant or not):
```
Unknown class _TtC8Kolibree20EmptyAppointmentView in Interface Builder file.
Unknown class _TtC8Kolibree23UpcomingAppointmentView in Interface Builder file.
```

**Expected outcome**
Carthage should not crash when testing on actual devices.

**Comment by Bawenang** [comment]
After more investigations, this happen on the custom UIViews that will be created on the fly when it is set as the class for an empty view. This is the code for the erroneous UIView:
```
class UpcomingAppointmentView: UIView {
    ...
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
        
        // Instantiate view and add as subview of the empty view.
        guard let nib = R.nib.upcomingAppointmentView(owner: self) else { return }
        addSubview(nib, method: .fill)
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()

        ...
    }
    
    func setup(with param: AppointmentCellParam) {
        self.param = param
        ...
    }
}
```

It is used as the custom class in an empty UIView in the storyboard like this:
<img width="258" alt="Screen Shot 2021-05-20 at 13 11 04" src="https://user-images.githubusercontent.com/6904722/118961533-9b7bab80-b98e-11eb-8d75-7c17bb47f683.png">
<img width="321" alt="Screen Shot 2021-05-20 at 13 10 52" src="https://user-images.githubusercontent.com/6904722/118961543-9e769c00-b98e-11eb-97a4-7d7b70654ed8.png">

When I added some breakpoints, I noticed that the `init?(coder:)` and `awakeFromNib()` are not called at all. But when I reverted back to using fat frameworks workaround, it worked just fine on both simulators and actual devices.
